### ISSUE TITLE
Style/AccessorGrouping cop autocorrect mess-up code with comments

### ISSUE THREAD

**Issue by ShockwaveNN** [rubocop-hq/rubocop#8289]
## Steps to reproduce the problem

1. Create `test.rb` file with content:
```ruby
class Foo
  # [String] Some bar value return
  attr_reader :bar

  # @return [String] value of foo_bar
  attr_reader :foo_bar
end

```

2. run `rubocop -a` with enabled `Style/AccessorGrouping` cop

## Expected behavior

Do not mess up my comments and `yard` documentation

## Actual behavior

Result file is:
```ruby
class Foo
  # [String] Some bar value return
  attr_reader :bar, :foo_bar

  # @return [String] value of foo_bar
end

```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.87.1 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.7.1 x86_64-linux)
```

**Comment by marcandre** [comment]
Good catch! We should consider this a bug and not raise an offense in this case.
